diff --git a/build_config.h b/build_config.h
index c7b02664c..a6863d2d4 100644
--- a/build_config.h
+++ b/build_config.h
@@ -173,7 +174,10 @@
 #define ARCH_CPU_BIG_ENDIAN 1
 #endif
 #else
-#error Please add support for your architecture in build/build_config.h
+//#error Please add support for your architecture in build/build_config.h
+#define ARCH_CPU_X86_FAMILY 1
+#define ARCH_CPU_32_BITS 1
+#define ARCH_CPU_LITTLE_ENDIAN 1 
 #endif
 
 // Type detection for wchar_t.
diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 93a97ff28..01e6f96d0 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -524,7 +524,7 @@ default_compiler_configs = [
   "//build/config/compiler:no_exceptions",
   "//build/config/compiler:no_rtti",
   "//build/config/compiler:runtime_library",
-  "//build/config/compiler:thin_archive",
+#  "//build/config/compiler:thin_archive",
   "//build/config/coverage:default_coverage",
   "//build/config/sanitizers:default_sanitizer_flags",
 ]
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 40696d584..68fb948be 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -702,8 +702,8 @@ config("compiler_cpu_abi") {
     # simplicity we always explicitly set the architecture.
     if (current_cpu == "x64") {
       cflags += [
-        "-m64",
-        "-march=x86-64",
+#        "-m64",
+#        "-march=x86-64",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
@@ -1415,7 +1415,7 @@ config("default_warnings") {
 
         # Ignore warnings about MSVC optimization pragmas.
         # TODO(thakis): Only for no_chromium_code? http://crbug.com/505314
-        "-Wno-ignored-pragma-optimize",
+        "-Wno-deprecated-register",
       ]
     } else if (use_xcode_clang) {
       cflags += [
@@ -2142,7 +2142,7 @@ config("symbols") {
         # which lld needs in order to generate a correct GDB index.
         # TODO(pcc): Try to make lld understand non-GNU-format pubnames
         # sections (llvm.org/PR34820).
-        cflags += [ "-ggnu-pubnames" ]
+#        cflags += [ "-ggnu-pubnames" ]
       }
       ldflags += [ "-Wl,--gdb-index" ]
     }
diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index b6f63d755..0ec05a8e0 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -605,8 +605,8 @@ template("clang_toolchain") {
 
   gcc_toolchain(target_name) {
     prefix = rebase_path("$clang_base_path/bin", root_build_dir)
-    cc = "$prefix/clang"
-    cxx = "$prefix/clang++"
+    cc = "emcc"
+    cxx = "em++"
     ld = cxx
     readelf = "${toolprefix}readelf"
     ar = "${prefix}/llvm-ar"
